src/app/Http/Controllers/Operations/FetchAjaxOperation.php

```diff
@@ -0,0 +1,159 @@
+    }
+
+    /**
+     * Gets items from database and returns to selects.
+     *
+     * @param Request $request
+     * @return void
+     */
+    public function fetchMultipleItems(Request $request)
+    {
+        $entityRoutes = $this->crud->getOperationSetting('ajaxEntities');
+        $routeSegment = $this->getRouteSegment($request->route()->uri);
+
+        $model = $entityRoutes[$routeSegment]['model'];
+
+        $instance = new $model;
+
+        $itemsPerPage = isset($entityRoutes[$routeSegment]['itemsPerPage']) ? $entityRoutes[$routeSegment]['itemsPerPage'] : 10;
+
+        //get searchable attributes if defined otherwise get identifiable attributes from model
+        $whereToSearch = isset($entityRoutes[$routeSegment]['searchableAttributes']) ?
+        $entityRoutes[$routeSegment]['searchableAttributes'] : $model::getIdentifiableName();
+
+        $table = Config::get('database.connections.'.Config::get('database.default').'.prefix').$instance->getTable();
+
+        $instanceKey = $instance->getKeyName();
+
+        $conn = $model::getPreparedConnection($instance);
+
+        if ($request->has('q')) {
+            if (empty($request->input('q'))) {
+                $search_term = false;
+            } else {
+                $search_term = $request->input('q');
+            }
+        }
+
+        $query = isset($entityRoutes[$routeSegment]['query']) ? $entityRoutes[$routeSegment]['query'] : $model;
+
+        if (is_callable($query)) {
+            $instance = $query($instance);
+        }
+
+        if (isset($search_term)) {
+            if ($search_term === false) {
+                return $instance->first();
+            }
+            foreach ($whereToSearch as $searchColumn) {
+                $columnType = $conn->getSchemaBuilder()->getColumnType($table, $searchColumn);
+
+                $operation = ! isset($isFirst) ? 'where' : 'orWhere';
+
+                if ($columnType == 'string') {
+                    $instance->{$operation}($searchColumn, 'LIKE', '%'.$search_term.'%');
+                } else {
+                    $instance->{$operation}($searchColumn, $search_term);
+                }
+                $isFirst = true;
+            }
+        }
+
+        $results = $instance->paginate($itemsPerPage);
+
+        return $results;
+    }
+
+    /**
+     * Fetches a single item from database.
+     *
+     * @param int $id
+     * @return void
+     */
+    public function fetchSingleItem($id)
+    {
+        $request = request()->instance();
+        $routeSegment = $this->getRouteSegment($request->route()->uri);
+        $entityRoutes = $this->crud->getOperationSetting('ajaxEntities');
+        $model = $entityRoutes[$routeSegment]['model'];
+
+        return $model::findOrFail($id);
+    }
+
+    /**
+     * Get url segment from uri.
+     *
+     * @param string $uri
+     * @return string
+     */
+    public function getRouteSegment($uri)
```

Comment:
Since this is a Controller trait, and developers are able to _easily_ overwrite this, I'd rather keep the amount of methods in Operation trait to a minimum. So instead of using having this method maybe we can use ```end(explode('/', $uri)``` everywhere. Or better yet, if we're only talking about the current request, we could do the more readable ```last(request()->segments())``` everywhere. What do you think @pxpm ?

Reply:
I agree. We should keep operation methods to the minimum to easy overwrite.